model/task/task.go

```diff
@@ -871,35 +884,240 @@ func (t *Task) ActivateTask(caller string) error {
 	t.ActivatedBy = caller
 	t.Activated = true
 	t.ActivatedTime = time.Now()
-	return UpdateOne(
+
+	return ActivateTasks([]Task{*t}, caller)
+}
+
+func ActivateTasks(tasks []Task, caller string) error {
+	taskIDs := make([]string, 0, len(tasks))
+	for _, t := range tasks {
+		taskIDs = append(taskIDs, t.Id)
+	}
+
+	_, err := UpdateAll(
 		bson.M{
-			IdKey: t.Id,
+			IdKey: bson.M{"$in": taskIDs},
 		},
 		bson.M{
 			"$set": bson.M{
 				ActivatedKey:     true,
 				ActivatedByKey:   caller,
-				ActivatedTimeKey: t.ActivatedTime,
+				ActivatedTimeKey: time.Now(),
```

Comment:
In the ActivateTask case, the local t.ActivatedTime and the DB ActivatedTime won't match, fyi.

Reply:
fixed